setup.py

```diff
@@ -27,14 +29,10 @@
     with open(req_file) as f:
         extras_require[e] = [line.strip() for line in f]
 
-# Loading version number
-with open(path.join(here, 'dowhy', 'VERSION')) as version_file:
-    version = version_file.read().strip()
-    print(version)
-
 setup(
     name='dowhy',
-    version=version,
+    version=versioneer.get_version(),
```

Comment:
This is cool. A few questions:
1. for a new release, where do we specify the new release version so that versioneer fills this correctly?
2.  for an update to the master docs, I'm guessing no change is needed in the version?

Reply:
> 1. for a new release, where do we specify the new release version so that versioneer fills this correctly?

You specify the version via `git tag`

See https://github.com/python-versioneer/python-versioneer/blob/master/INSTALL.md#post-installation-usage, which says:
> If you distribute your project through PyPI, then the release process should boil down to two steps:
>
> - 1: git tag 1.0
> - 2: python setup.py register sdist upload

> 2. for an update to the master docs, I'm guessing no change is needed in the version?

Correct. Pushing to `master`/`main` will modify the documentation, whereas the documentation for a specific release is frozen.

It's important to realize that this gives us slightly less flexibility in terms of updating the docs for an already _released_ version. The only way I see how this could work is by creating a branch on the released-tagged commit and retagging a new commit with the same tag (basically moving the tag to another commit and forcing-pushing that change). It's a bit dirty, but would probably work. I'd prefer a more aggressive release frequency though.

Reply:
interesting, this automation will be useful. just to confirm: previously, I used to update the version manually in dowhy/version.py, and then create a release using the github UI (that created a tag and automatically [pypi publish workflow](https://github.com/py-why/dowhy/blob/main/.github/workflows/python-publish.yml)).

Now, the steps will be:
1. create the release using the github UI (which also creates a tag)
2. let the pypi publish workflow run automatically.

In this case, would versioneer automatically pick the latest version number?

For older docs: As I mentioned on the other PR, I think it is okay to keep the docs only for v0.7 or v0.8 onwards. They are backwards-compatible and we can encourage people to update.

Reply:
> interesting, this automation will be useful. just to confirm: previously, I used to update the version manually in dowhy/version.py, and then create a release using the github UI (that created a tag and automatically [pypi publish workflow](https://github.com/py-why/dowhy/blob/main/.github/workflows/python-publish.yml)).
> 
> Now, the steps will be:
> 
> 1. create the release using the github UI (which also creates a tag)
> 2. let the pypi publish workflow run automatically.
> 
> In this case, would versioneer automatically pick the latest version number?

Yes. Internally, versioneer uses a specially parameterized `git describe` invocation to "heuristically" find the _latest release_ tag. And I assume when GitHub UI creates the version tag, it will have to push this tag and that push will also trigger the GitHub action to build and deploy the docs and then versioneer will pick this up.

> For older docs: As I mentioned on the other PR, I think it is okay to keep the docs only for v0.7 or v0.8 onwards. They are backwards-compatible and we can encourage people to update.

Makes sense.

Reply:
Oh, wait a second. Let me actually check the other workflows. To see how this would work. Sorry, I didn't realize PyPI publication also happens through a workflow.

Reply:
Okay, yes, this should work. The moment you use the GitHub UI to create the release and the git tag, it will push this tag and that will trigger the publish workflow which then just relies on the usual `python setup.py sdist bdist_wheel` call which uses versioneer to choose the current/latest version from the just-created git tag.